### PR TITLE
ARP Part 6: Added shared rule

### PR DESCRIPTION
**Pull request by niksv**
Part of #3118

### CODE REVIEW

File: src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/command/flow/InstallIngressFlow.java [hunk 1]
```diff
@@ -143,6 +154,7 @@ public InstallIngressFlow(@JsonProperty(TRANSACTION_ID) final UUID transactionId
         setMeterId(meterId);
         setEnableLldp(enableLldp);
         setEgressSwitchId(egressSwitchId);
+        this.enableArp = enableArp;
```
Comment (timofei-durakov) [inline src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/command/flow/InstallIngressFlow.java]: nit: align it with lldp(there is a setter for the lldp above)

File: src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/command/flow/InstallOneSwitchFlow.java [hunk 2]
```diff
@@ -124,14 +135,16 @@ public InstallOneSwitchFlow(@JsonProperty(TRANSACTION_ID) final UUID transaction
...
+                                @JsonProperty("enable_lldp") final boolean enableLldp,
+                                @JsonProperty("enable_arp") final boolean enableArp) {
         super(transactionId, id, cookie, switchId, inputPort, outputPort, multiTable);
         setInputVlanId(inputVlanId);
         setOutputVlanId(outputVlanId);
         setOutputVlanType(outputVlanType);
         setBandwidth(bandwidth);
         setMeterId(meterId);
         setEnableLldp(enableLldp);
+        this.enableArp = enableArp;
```
Comment (timofei-durakov) [inline src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/command/flow/InstallOneSwitchFlow.java]: same as above

File: src-java/floodlight-service/floodlight-modules/src/test/java/org/openkilda/floodlight/command/flow/ingress/IngressFlowSegmentCommandJsonTest.java [hunk 3]
```diff
@@ -56,7 +56,7 @@ protected IngressFlowSegmentRequest makeRequest() {
                 new SwitchId(20),
                 8,
                 new FlowTransitEncapsulation(9, FlowEncapsulationType.TRANSIT_VLAN),
-                new RemoveSharedRulesContext(false, false));
+                new RemoveSharedRulesContext(false, false, false));
```
Comment (timofei-durakov) [inline src-java/floodlight-service/floodlight-modules/src/test/java/org/openkilda/floodlight/command/flow/ingress/IngressFlowSegmentCommandJsonTest.java]: nit: maybe switch to builder? or create final constants of remove shared rules context instance

File: src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/command/flow/InstallIngressFlow.java (hunk 1 above)
Comment (niksv) [reply src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/command/flow/InstallIngressFlow.java]: fixed

File: src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/command/flow/InstallOneSwitchFlow.java (hunk 2 above)
Comment (niksv) [reply src-java/floodlight-service/floodlight-api/src/main/java/org/openkilda/messaging/command/flow/InstallOneSwitchFlow.java]: fixed

File: src-java/floodlight-service/floodlight-modules/src/test/java/org/openkilda/floodlight/command/flow/ingress/IngressFlowSegmentCommandJsonTest.java (hunk 3 above)
Comment (niksv) [reply src-java/floodlight-service/floodlight-modules/src/test/java/org/openkilda/floodlight/command/flow/ingress/IngressFlowSegmentCommandJsonTest.java]: fixed

**Pull request merged**